BUG: Make building docs with missing python3 easier

### Describe the issue:

I've spent quite a few minutes trying to decrypt why I cannot build the docs locally, when local testing, etc all work fine, dependencies are installed, etc. Naturally, I should have started with grepping for the error message to discover that I should do `make PYTHON=python html` instead.

Ideally, it would be nice if the machinery wouldn't assume python3 as the name of the executable (are there any other python in these days?), or at minimum would give a useful error message (similar to the one given with mismatching version number).




### Reproduce the code example:

```shell
make  html
```

When no `python3` executable is available, or it's outside of the current virtualenv/pyenv


### Error message:

```shell
numpy not found, cannot build documentation without successful "import numpy"
make: *** [version-check] Error 1
```

--- a/doc/Makefile
+++ b/doc/Makefile
@@ -88,6 +88,7 @@ ifeq "$(GITVER)" "Unknown"
 	# @echo sdist build with unlabeled sources
 else ifeq ("", "$(NUMPYVER)")
 	@echo numpy not found, cannot build documentation without successful \"import numpy\"
+	@echo Try overriding the makefile PYTHON variable with '"make PYTHON=python $(MAKECMDGOALS)"'
 	@exit 1
 else ifneq ($(NUMPYVER),$(GITVER))
 	@echo installed numpy $(NUMPYVER) != current repo git version \'$(GITVER)\'
@@ -263,4 +264,3 @@ info:
 
 show:
 	@python -c "import webbrowser; webbrowser.open_new_tab('file://$(PWD)/build/html/index.html')"
-

